flink/src/test/java/org/apache/iceberg/flink/TestCatalogTableLoader.java

```diff
@@ -0,0 +1,107 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.hadoop.HadoopTables;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.types.Types;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Test for {@link CatalogLoader} and {@link TableLoader}.
+ */
+public class TestCatalogTableLoader extends FlinkTestBase {
+
+  private static File warehouse = null;
+  private static final TableIdentifier IDENTIFIER = TableIdentifier.of("default", "my_table");
+  private static final Schema SCHEMA = new Schema(Types.NestedField.required(1, "f1", Types.StringType.get()));
+
+  @BeforeClass
+  public static void createWarehouse() throws IOException {
+    warehouse = File.createTempFile("warehouse", null);
+    Assert.assertTrue(warehouse.delete());
+    hiveConf.set("my_key", "my_value");
+  }
+
+  @AfterClass
+  public static void dropWarehouse() {
+    if (warehouse != null && warehouse.exists()) {
+      warehouse.delete();
+    }
+  }
+
+  @Test
+  public void testHadoopCatalogLoader() throws IOException, ClassNotFoundException {
+    CatalogLoader loader = CatalogLoader.hadoop("my_catalog", hiveConf, "file:" + warehouse);
+    validateHadoopConf(javaSerAndDeSer(loader).loadCatalog().createTable(IDENTIFIER, SCHEMA));
+  }
+
+  @Test
+  public void testHiveCatalogLoader() throws IOException, ClassNotFoundException {
+    CatalogLoader loader = CatalogLoader.hive("my_catalog", hiveConf, null, 2);
+    validateHadoopConf(javaSerAndDeSer(loader).loadCatalog().createTable(IDENTIFIER, SCHEMA));
+  }
+
+  @Test
+  public void testHadoopTableLoader() throws IOException, ClassNotFoundException {
+    String location = "file:" + warehouse + "/my_table";
+    new HadoopTables(hiveConf).create(SCHEMA, location);
+    TableLoader loader = TableLoader.fromHadoopTable(location, hiveConf);
+    TableLoader copied = javaSerAndDeSer(loader);
+    copied.open();
+    try {
+      validateHadoopConf(copied.loadTable());
+    } finally {
+      copied.close();
+    }
+  }
+
```

Comment:
Missed a test for `HiveTableLoader` ?  I think we don't have to open the table actually,  just need to ensure the members inside the instance are the same, maybe adding an equals inside the CatalogLoader or TableLoader.

Reply:
You mean `Tableloader.CatalogTableloader` class? I think we don't need to add case for it since we have case for `Catalog Loader`s.
Opening the table can cover more since they are loaders, we can try to load something and ensure that is correct.

Reply:
OK, it's 'CatalogTableLoader'.  it has two members: `CatalogLoader` & `String identifier` (the other one is transient member). I agree that in this case if we could assert that `CatalogLoader` is serializing & deserializing correctly, then it imply that `CatalogTableLoader` should be OK. But providing those unit tests is to ensure that we won't break this point in future.  What if others introduce another non-serializable member in `CatalogTableLoader` class?  that's why I said it seems to be more reasonable to write a ser/der unit test for `CatalogTableLoader`.

Reply:
make sense, I'll add test.